### ISSUE TITLE
ValueError: invalid mode: 'rU' while trying to load binding.gyp when installing in docker node alpine

### ISSUE THREAD

**Issue by deemaagog** [Automattic/node-canvas#2325]
## Issue or Feature
- [x] If this is an issue with installation, I have read the [troubleshooting guide](https://github.com/Automattic/node-canvas/issues/1511).

Getting error **ValueError: invalid mode: 'rU' while trying to load binding.gyp** when building node-canvas from source in docker. Log says 
` gyp info find Python using Python version 3.11.6 found at "/usr/bin/python"`
As far as I understand I need python 3.10, but I cant figure out where 3.11.6 comes from since there is no preinstalled python in node alpine image. Ideas?

My Dockerfile starts as 

```
FROM node:18-alpine as builder

RUN apk add --no-cache build-base g++ cairo-dev jpeg-dev pango-dev giflib-dev jq

WORKDIR /usr/src

COPY package*.json ./
COPY yarn.lock ./


COPY ./packages/common ./packages/common
COPY ./packages/backend ./packages/backend

RUN yarn install -> Error here

```

## Your Environment
* Version of node-canvas (output of `npm list canvas` or `yarn list canvas`): 2.11.2
* Environment (e.g. node 20.9.0 on macOS 14.1.1): docker node-18:alpine

**Comment by zbjornson** [comment]
It looks like this was fixed in node-gyp v8.0.0 (https://github.com/nodejs/node-gyp/commit/9e1397c52e429eb96a9013622cffffda56c78632#diff-787d31707936c8991eda641b83272ec8d1896fa433f05a2712669ceaa6808481). Try `npm install -g node-gyp@latest`.

**Comment by deemaagog** [comment]
So, that was extremely old lerna version that caused old  node-gyp to be in my node_modules. Thanks a lot for pointing me in that direction!
